src/inc/gcc_uuid.h

```diff
@@ -0,0 +1,28 @@
+#ifdef __cplusplus
+
+typedef struct _GUID {          // size is 16
+    unsigned int    Data1;
+    unsigned short  Data2;
+    unsigned short  Data3;
+    unsigned char   Data4[8];
+} GUID;
+
+#define GUID_DEFINED
+
+typedef GUID IID;
+
+class TypeToIID
+{
+public:
+    // Default version of the GetIID function for types where the type was not bound to IID using the macro below
+    template<typename TInterface>
+    static IID GetIID();
+};
+
+
+#define BIND_UUID_OF(T) struct T; template<> IID TypeToIID::GetIID<T>() { return IID_##T; }
```

Comment:
I think this should take `type` and `typePrefix`, instead of just `T` to form:

```c
 #define BIND_UUID_OF(T) struct type; template<> IID TypeToIID::GetIID<type>() { return typePrefix_##T; }
```

In edge case scenario, type and typePrefix may differ, e.g. in `cordebug_i.cpp`:

```c
MIDL_DEFINE_GUID(IID, LIBID_CORDBLib,0x53D13620,0xF417,0x11d1,0x97,0x62,0xA6,0x38,0x26,0xA4,0xF2,0x55);
```